pkg/prowgen/jobbase.go

```diff
@@ -131,13 +131,19 @@ func NewProwJobBaseBuilderForTest(configSpec *cioperatorapi.ReleaseBuildConfigur
 		if configSpec.Releases != nil {
 			p.PodSpec.Add(CIPullSecret())
 		}
+		if clusterType := test.MultiStageTestConfigurationLiteral.ClusterProfile.ClusterType(); clusterType != "" {
```

Comment:
We should consider the env var as well.

Reply:
Unrelated to the envvar, could we piggyback on the `if test.MultiStageTestConfigurationLiteral.ClusterProfile != ""` condition above? Like:

```go
if clusterProfile := test.MultiStageTestConfigurationLiteral.ClusterProfile; clusterProfile != "" {
  p.PodSpec.Add(ClusterProfile(clusterProfile, test.As), LeaseClient())
  p.WithLabel(cioperatorapi.CloudLabel, clusterProfile.ClusterType())
```

I guess this is only possible if all possible clsuter profile values have a non-empty `ClusterType()`